### PR TITLE
[JENKINS-69608] Fix 'Notification URL' dropdown

### PR DESCRIPTION
**Pull request by frankie139506**
See [JENKINS-69608](https://issues.jenkins.io/browse/JENKINS-69608).

### Proposed changelog entries

- Make UI more reasonable.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  - Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [X] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content-Security-Policy directives (see [documentation on jenkins.io](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7109"></a>

### CODE REVIEW

File: core/src/main/resources/lib/form/entry.jelly [hunk 1]
```diff
@@ -97,7 +97,7 @@ THE SOFTWARE.
       <j:otherwise>
         <j:choose>
           <j:when test="${attrs.help!=null}">
-            <div class="setting-main help-sibling" style="display: inline-flex; align-items: center;">
```
Comment (timja) [inline core/src/main/resources/lib/form/entry.jelly:100]: checking this in an isolated way this does appear to fix it, but need to check if it impacts other components / why this was done

**Review by NotMyFault — COMMENTED** [review]
Is there a particular reasoning, why this should be addressed in core? To me, this looks like an old dropdown format part of display api plugin, and therefore should be addressed there.

**Comment by timja** [comment]
> Is there a particular reasoning, why this should be addressed in core? To me, this looks like an old dropdown format part of display api plugin, and therefore should be addressed there.

It's just using core controls:
https://github.com/jenkinsci/display-url-api-plugin/blob/cfe07af98c1c37efa0ffaa3de4133757f92ec732/src/main/resources/org/jenkinsci/plugins/displayurlapi/user/PreferredProviderUserProperty/config.jelly#L4-L8

Could try removing the `f:block` to see if it fixes it

**Comment by timja** [comment]
I think it's because there's no title on the entry.

Given how rare that is this is probably okay

**Comment by timja** [comment]
Looking at other components on that page, they all specify a label. So I would recommend that as a fix

**Comment by NotMyFault** [comment]
> Looking at other components on that page, they all specify a label. So I would recommend that as a fix

Thought so, thanks for the clarification!

File: core/src/main/resources/lib/form/entry.jelly (hunk 1 above)
Comment (NotMyFault) [reply core/src/main/resources/lib/form/entry.jelly:101]: I was looking around on other occasions and this seems to have no impact.

**Review by NotMyFault — APPROVED** [review]

File: core/src/main/resources/lib/form/entry.jelly (hunk 1 above)
Comment (timja) [reply core/src/main/resources/lib/form/entry.jelly:101]: it moves the help icon to the next line.

**Review by timja — CHANGES_REQUESTED** [review]
Closing as no activity in over a year.

I've reviewed it again today and my recommendation stands that a label is required here.
I've filed a PR on the display url api plugin itself to fix it

https://github.com/jenkinsci/display-url-api-plugin/pull/206

**Pull request closed without merging**